app/src/main/java/org/mifos/mobilebanking/ui/fragments/AccountsFragment.java

```diff
@@ -299,29 +299,77 @@ public void showEmptyAccounts(String emptyAccounts) {
      * recyclerview.
      * @param input String which is needs to be searched in list
      */
-    public void searchSavingsAccount(String input) {
+    public void searchSavingsAccount(String input, List<CheckboxStatus> statusModelList) {
+        List<SavingAccount> savingAccountSearchList = savingAccounts;
+        if (statusModelList != null) {
+            for (CheckboxStatus statusList: statusModelList) {
+                if (statusList.component3()) {
+                    List<SavingAccount> filteredSavings = new ArrayList<>();
+                    for (CheckboxStatus status :
+                            accountsPresenter.getCheckedStatus(statusModelList)) {
+                        filteredSavings.addAll(
+                                accountsPresenter.getFilteredSavingsAccount(savingAccounts,
+                                        status));
```

Comment:
We can minimize the code lines here. We can remove all the elements of `savingAccountSearchList` and then add the elements here.